lib/context_adapter_helper.js

```diff
@@ -0,0 +1,724 @@
+        message.version = caPackage.version;
+      }
+    }
+    if (Object.getOwnPropertyNames(message).length === 0) {
+      message.version = 'No version information available';
+    }
+    return message;
+  }
+
+  /**
+   * Sends a queryContext request to the configured Context Broker
+   * @param {string} service Service header to be used in the request
+   * @param {string} servicePath Service Path header to be used in the request
+   * @param {object} payload Payload to be included in the request
+   * @param {function} callback Callback to be called informing of any errors or returning
+   *  the obtained response
+   */
+  function sendQueryContext(service, servicePath, payload, callback) {
+    request({
+      method: 'POST',
+      uri: 'http://' + caConfig.CB_HOST + ':' + caConfig.CB_PORT + caConfig.CB_PATH + '/queryContext',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'Fiware-Service': service || caConfig.SERVICE,
+        'Fiware-ServicePath': servicePath || caConfig.SERVICE_PATH
+      },
+      json: true,
+      body: payload
+    }, callback);
+  }
+
+  /**
+   * Sends a updateContext request to the configured Context Broker
+   * @param {string} service Service header to be used in the request
+   * @param {string} servicePath Service Path header to be used in the request
+   * @param {object} payload Payload to be included in the request
+   * @param {function} callback Callback to be called informing of any errors or returning
+   *  the obtained response
+   */
+  function sendUpdateContext(service, servicePath, payload, callback) {
+    request({
+      method: 'POST',
+      uri: 'http://' + caConfig.CB_HOST + ':' + caConfig.CB_PORT + caConfig.CB_PATH + '/updateContext',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'Fiware-Service': service || caConfig.SERVICE,
+        'Fiware-ServicePath': servicePath || caConfig.SERVICE_PATH
+      },
+      json: true,
+      body: payload
+    }, callback);
+  }
+
+  /**
+   * Indicates if certain attribute is included in an attributes array
+   * @param attributes The attributes array
+   * @param attributeName The attribute name to find
+   * @return {boolean} The result of the search
+   */
+  function isAttributeIncluded(attributes, attributeName) {
+    for (var i = 0; i < attributes.length; i++) {
+      if (attributes[i].name && attributes[i].name === attributeName) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Checks that an updateContext request sent to the Context Adapter is well-formed
+   * @param {object} request The request sent to the Context Adapter
+   * @param {function} callback Callback to be called with the result of the check
+   */
+  function checkUpdateContextRequest(request, callback) {
+    /*
+      A valid updateContext payload is one such as the following one:
```

Comment:
Don't give such big examples inside the code, as it tends to make it less readable. Use a section in the documentation instead if you think is appropriate.